daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -144,14 +130,6 @@ class InvalidParserException(msg: String, cause: Throwable = null) extends Excep
 
 class Compiler private (var validateDFDLSchemas: Boolean,
   var tunables : DaffodilTunables,
```

Comment:
Can we remove the `var`s here too?

Reply:
When you say remove do you want them removed completely or see if they can be `val`?

Reply:
In these cases, we can just remove the `var` keyword and scala treats them as private vals. We still need to keep them as constructor parameters.

Reply:
Okay that makes sense, so for these ones you had asked about should I remove the `private` keyword or keep it for ones that have it?

Reply:
I believe `private var` can be removed and it is implied to be private val.

Reply:
Makes sense, do the changes just pushed look good?

Reply:
Looks reasonable to me :+1: